ext/include/opentelemetry/ext/http/client/curl/http_client_curl.h

```diff
@@ -153,6 +153,33 @@ class Session : public http_client::Session
     });
   }
 
+  virtual std::unique_ptr<http_client::Response> SendRequestSync(
+      http_client::SessionState &session_state) noexcept override
+  {
+    is_session_active_ = true;
+    std::string url    = host_ + std::string(http_request_->uri_);
+    curl_operation_.reset(new HttpOperation(http_request_->method_, url, nullptr, RequestMode::Sync,
+                                            http_request_->headers_, http_request_->body_, false,
+                                            http_request_->timeout_ms_));
+    curl_operation_->SendSync();
+    session_state = curl_operation_->GetSessionState();
+    if (curl_operation_->WasAborted())
+    {
+      session_state = http_client::SessionState::Cancelled;
+    }
+    if (curl_operation_->GetResponseCode() >= CURL_LAST)
+    {
+      // we have a http response
+      auto response      = std::unique_ptr<Response>(new Response());
+      response->headers_ = curl_operation_->GetResponseHeaders();
```

Comment:
How do you propagate Response status code? If you are using int for that, then in Java it'd typically allocate `-1` for failed response status code, which can then be used to determine if request was actually processed, as in connected and got some response code from server.

Reply:
Currently we use `uint16_t;` for status_code